### DISCUSSION TITLE
How to use raw sql or sql query builder in *.svelte files?

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#3313] [Q&A] [answered]
Hi, I have two collections: students and sponsors

"students" has a multi-relation field that links to "sponsors"

![image](https://github.com/pocketbase/pocketbase/assets/12494537/dab76e82-f7d6-492f-a35c-3c74f8fd5292)

![image](https://github.com/pocketbase/pocketbase/assets/12494537/c46d6865-3e7e-4005-9555-042c57e2122d)

I understand back-relation expansion is not available (hopefully it will be in the future).  To workaround this I wish to have a intermediary relation table created that will implement the many-to-many relation between "students" and "sponsors". But I can't find any documentation about using web api calls for this. Correct me if I'm wrong - I will have to use raw sql to achieve create this many-to-many relation. 

I can't seem to figure out how to access $app.dao() in my *.svelte files. Could any please point me in the right direction with this or even suggest an alternative solution to the overall problem I have?  Many thanks in advance.

P.S. I know this relates to svelte as well but I thought of trying my luck asking this question here first.

**Comment by ganigeorgiev** [comment]
I'm failing to understand what you are trying to do.

Which relation do you want to expand?

The JS `pb_hooks` can't be executed from within Svelte (or any other JS environment).
They can be executed **only** from `.pb.js` files inside your `pb_hooks` directory (see https://pocketbase.io/docs/js-overview/). You can [register a custom route](https://pocketbase.io/docs/js-routing/#registering-new-routes) and call that route using the [SDK `pb.send()`](https://pocketbase.io/docs/js-routing/#sending-request-to-custom-routes-using-the-sdks), but again, I'm not sure if that is exactly what you are trying to do.

**Reply by rameshkumar70** [reply]
Thank you for the lightning speed reply. I wish to expand in both ways actually.....I'd like to get a student with all her sponsors, and separately, a sponsor with all his students. 

Where do I create the pb_hooks directory? Is it in the same directory where the pockebase executable resides?

**Reply by ganigeorgiev** [reply, marked as answer]
If you create a junction collection, eg. `studentSponsor` with 2 single `relation` fields pointing back to the `students` and `sponsors`, then you can expand it both ways (_as long as the client is allowed to view the expanded records_):

```js
await pb.collection("students").getList(1, 30, {
    expand: "studentSponsor(student).sponsor"
})

await pb.collection("sponsors").getList(1, 30, {
    expand: "studentSponsor(sponsor).student"
})
```
_(see also see https://pocketbase.io/docs/working-with-relations/#back-relation-expand)_

---

> Where do I create the pb_hooks directory? Is it in the same directory where the pockebase executable resides?

Yes, you can create the directory next to the executable. Alternatively, you can also explicitly specify a different directory path using the `--hooksDir` flag.

**Comment by rameshkumar70** [comment]
You sir, are a godsend.....firstly for creating this project and making it accessible with beautiful utilities,  and more importantly, being here to respond to queries so promptly.
What you say makes sense now. I didn't know about the junction collection or rather, it didn't strike me as a "relations table". I will try this and also the pb_hooks suggestion. Thank you so much once again.
